networks.Rmd

```diff
@@ -0,0 +1,259 @@
+
+### Conversion
+
+It is often that network data is presented in a range of different formats depending on where you get it from. tidygraph understands most of the different classes used in R for network data and these can be converted using `as_tbl_graph()`. Below is an example of both a converting a data frame encoding an edgelist, as well as converting the result of `hclust()`
+
+```{r}
+data(highschool, package = "ggraph")
+head(highschool)
+```
+
+```{r}
+hs_graph <- as_tbl_graph(highschool, directed = FALSE)
+hs_graph
+```
+
+```{r}
+luv_clust <- hclust(dist(luv_colours[, 1:3]))
+luv_graph <- as_tbl_graph(luv_clust)
+luv_graph
+```
+
+We can see that tidygraph automatically adds additional info when converting, e.g. the *year* column in the highschool data, and the *label* and *leaf* property of the nodes in the hierarchical clustering.
+
+### Algorithms
+
+While simply manipulating networks is nice, the real benefit of networks come from the different operations that can be performed on them using the underlying structure. tidygraph has rich support for a range of different groups of algorithms such as centrality calculation (which node is most central), ranking (order nodes so nodes are located close to those they are connected to), grouping (finding clusters inside the network), etc. The algorithm API is designed to be used inside `mutate()` and will always return a vector with length and order matching the nodes or edges. Further, it does not require you to specify the graph or nodes you want to calculate for since this is given implicitly in the `mutate()` call. As an example, we will calculate the centrality of the nodes in our sample graph using the pagerank algorithm and sort the nodes according to that:
+
+```{r}
+graph %>% 
+  activate(nodes) %>% 
+  mutate(centrality = centrality_pagerank()) %>% 
+  arrange(desc(centrality))
+```
+
+### Want more?
+
+This is just a brief glimpse into tidygraph, for the sake of understanding ggraph. If you are interested in learning more, the tidygraph website gives an overview of all the functionalities in the package: <https://tidygraph.data-imaginist.com>
+
+## Visualizing networks
+
+ggraph builds on top of tidygraph and ggplot2 to allow a complete and familiar grammar of graphics for network data. Still, it is a little different from most ggplot2 extension packages since it works with another data type that is fundamentally different from tabular data. More so, most network visualizations don't concern themselves with mapping variables to `x` and `y` aesthetics since they are concerned with shoving the network topology more than relations between two variables. In order to show network topology the concept of layouts are employed. Layouts are algorithms that use the network structure to calculate (often arbitrary) `x` and `y` values for each node that can then be used for visualization purpose.
```

Comment:
I think I'd emphasise that with tabular data `x` and `y` are almost always mapped to existing variables, but with network data, `x` and `y` are almost always generated from the network topology.